source/MaterialXRender/Handlers/ImageHandler.h

```diff
@@ -3,60 +3,173 @@
+/// for a given target (such as a given shading language).
 ///
 class ImageHandler
 {
-public:
-    /// Default constructor
-    ImageHandler() {}
+  public:
+    /// Constructor. Assume at least one loader must be supplied.
+    ImageHandler(ImageLoaderPtr imageLoader)
+    {
+        _imageLoaders.push_back(imageLoader);
+    }
+
+    /// Add additional image loaders. Useful to handle different file
+    /// extension
+    /// @param loader Loader to add to list of available loaders.
+    void addLoader(ImageLoaderPtr loader)
+    {
+        _imageLoaders.push_back(loader);
+    }
     
     /// Default destructor
     virtual ~ImageHandler() {}
 
     /// Save image to disk. This method must be implemented by derived classes.
+    /// The first image loader which supports the file name extension will be used.
     /// @param fileName Name of file to save image to
-    /// @param width Width of image in pixels
-    /// @param height Height of image in pixels
-    /// @param channelCount Number of channels per pixel
-    /// @param buffer Floating point buffer of pixels.
+    /// @param imageDesc Description of image
     /// @return if save succeeded
     virtual bool saveImage(const std::string& fileName,
-                            unsigned int width,
-                            unsigned int height,
-                            unsigned int channelCount,
-                            const float* buffer) = 0;
+                           const ImageDesc &imageDesc);
 
-    /// Load an image from disk. This method must be implemented by derived classes.
-    /// @param fileName Name of file to load image from
-    /// @param width Width of image in pixels
-    /// @param height Height of image in pixels
-    /// @param channelCount Number of channels per pixel
-    /// @param buffer Floating point buffer of pixels.
+    /// Get an image from disk. This method must be implemented by derived classes.
+    /// The first image loader which supports the file name extension will be used.
+    /// @param fileName Name of file to load image from.
+    /// @param imageDesc Description of image updated during load.
+    /// @param generateMipMaps Generate mip maps if supported.
     /// @return if load succeeded
-    virtual bool loadImage(const std::string& fileName,
-                            unsigned int& width,
-                            unsigned int& height,
-                            unsigned int& channelCount,
-                            float** buffer) = 0;
-
-    /// Utility to create a default image if a given image cannot be loaded
-    /// The image should contain four channels (RGBA) with each channel
-    /// being unsigned char in size.
-    /// @param width Width of image in pixels
-    /// @param height Height of image in pixels
-    /// @param buffer Unsigned char buffer of pixels. 
+    virtual bool getImage(std::string& fileName, ImageDesc& desc, bool generateMipMaps);
```

Comment:
The saveImage/loadImage naming is used on the ImageLoaded classes, so maybe change back to loadImage here for consistency.

Reply:
I may change both to "acquire" instead.